src/controllers/contact.ts

```diff
@@ -15,6 +27,27 @@ export const postContact = asyncHandler(
       throw new UnprocessableEntity({ errors: errors.array() })
     }
 
-    throw new NotImplemented()
+    try {
+      const templateOptions = {
+        name: name,
+        message: req.body.message,
+      }
+
+      const htmlTemplate = await mail.renderTemplate(MailerTemplate.contactUsHtml, templateOptions)
+      const txtTemplate = await mail.renderTemplate(MailerTemplate.contactUsTxt, templateOptions)
+
+      await mail.sendMail({
+        from: email,
+        to: 'support@foobar.com',
+        subject: 'FooBar Support',
+        html: htmlTemplate,
+        text: txtTemplate,
+      })
```

Comment:
Once you have the `mail.render` method you can remove these keys and just use spread operate when calling it:
```suggestion
      await mail.sendMail({
        from: email,
        to: 'support@foobar.com',
        subject: 'FooBar Support',
        ...mail.render(MailTemplate.contactUs, context),
      })
```

Reply:
Complete